sympy/sets/sets.py

```diff
@@ -964,6 +964,9 @@ def _boundary(self):
         return FiniteSet(self.start, self.end)
 
     def _contains(self, other):
+        if isinstance(other, Set):
```

Comment:
I think this is too restrictive. *Anything* that won't compare in a Boolean should return false, perhaps? So everything after the test `other.is_real...` could be wrapped in a `try/except` to catch a bad boolean arg. But...Booleans aren't detecting bad args well enough:

This is ok:

```
>>> I<2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\core\expr.py", line 299, in __lt__
    raise TypeError("Invalid comparison of complex %s" % me)
TypeError: Invalid comparison of complex I
```

This isn't:

```
>>> Interval(1,2)<3
False
>>> FiniteSet(2)<3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy\sets\sets.py", line 1996, in __lt__
    return self.is_proper_subset(other)
  File "sympy\sets\sets.py", line 361, in is_proper_subset
    raise ValueError("Unknown argument '%s'" % other)
ValueError: Unknown argument '3'
```

So perhaps you should just return False if other is not an Expr:

```
>>> isinstance(S(2), Expr)
True
>>> isinstance(x, Expr)
True
```